CODE_STYLE.md

```diff
@@ -0,0 +1,943 @@
+    summary in the first line and eventually, go to greater lengths in
+    your message's body. We also like commit message with a **type** and **scope**.
+
+  - Please create commits containing **related changes**. For example, two different bugfixes
+    should produce two separate commits. A feature should be made of commits splitted by
+    **logical chunks** (no half-done changes). Use your best judgement as to how many commits
+    your changes require.
+
+  Each commit message consists of a **header**, a **body** and a **footer**.  The header has a special
+  format that includes a **type**, a **scope** and a **subject**:
+
+  ```
+  <type>(<scope>): <subject>
+  <BLANK LINE>
+  <body>
+  <BLANK LINE>
+  <footer>
+  ```
+
+  The **header** is mandatory. Any line of the commit message cannot be longer
+  100 characters! This allows the message to be easier to read on GitHub as well
+  as in various git tools.
+
+  #### Type
+  
+  Must be one of the following:
+
+  * **feat**: A new feature
+  * **fix**: A bug fix (based of the `next` branch)
+  * **hotfix**: A bug fix (based of the `master` branch)
+  * **docs**: Documentation only changes
+  * **style**: Changes that do not affect the meaning of the code (white-space, formatting, missing
+    semi-colons, etc)
+  * **refactor**: A code change that neither fixes a bug nor adds a feature
+  * **perf**: A code change that improves performance
+  * **test**: Adding missing or correcting existing tests
+  * **chore**: Changes to the build process or auxiliary tools and libraries such as documentation
+    generation
+
+  #### Scope
+  
+  The scope could be anything specifying the place of the commit change. Common examples:
+  
+  * **`<plugin-name>`**: This could be `basic-auth`, or `ldap` for example
+  * **api**: the RESTful management api
+  * **proxy**: the core proxy functionality
+  * **balancer**: the loadbalancer
+  * **dns**: dns resolution
+  * **dao**: datastore persistence functionality
+  * **cache**: for Lua or shm based caching
+  * **cli**: command line interface
+  * **deps**: for updating dependencies
+  * **conf**: for configuration related changes
+  
+  You can use `*` when the change affects more than a single scope.
+
+  ### Subject
+
+  The subject contains succinct description of the change:
+
+  * use the imperative, present tense: "change" not "changed" nor "changes"
+  * don't capitalize first letter
+  * no dot (.) at the end
+
+  ### Body
+
+  Just as in the **subject**, use the imperative, present tense: "change" not "changed" nor "changes".
+  The body should include the motivation for the change and contrast this with previous behavior.
+
+  ### Footer
+
+  The footer should contain any information about **Breaking Changes** and is also the place to
+  [reference GitHub issues that this commit closes](https://help.github.com/articles/closing-issues-via-commit-messages/).
```

Comment:
We should maybe include such an example of referencing to a GH issue, since not everyone might know how to reference them:
```
Fix #1234
```

And for another repo on GitHub:
```
Fix Mashape/vagrant-kong#1234
```